pkg/scheduler/framework/plugins/interpodaffinity/interpod_affinity_test.go

```diff
@@ -0,0 +1,1896 @@
+					Spec: v1.PodSpec{
+						NodeName: "nodeA",
+					},
+				},
+			},
+			nodes: []v1.Node{
+				{ObjectMeta: metav1.ObjectMeta{Name: "nodeA", Labels: map[string]string{"region": "r1", "zone": "z1", "hostname": "nodeA"}}},
+				{ObjectMeta: metav1.ObjectMeta{Name: "nodeB", Labels: map[string]string{"region": "r1", "zone": "z1", "hostname": "nodeB"}}},
+			},
+			wantStatuses: []*framework.Status{nil, nil},
+			name:         "Test incoming pod's affinity: firstly check if all affinityTerms match, and then check if all topologyKeys match",
+		},
+		{
+			pod: &v1.Pod{
+				Spec: v1.PodSpec{
+					Affinity: &v1.Affinity{
+						PodAffinity: &v1.PodAffinity{
+							RequiredDuringSchedulingIgnoredDuringExecution: []v1.PodAffinityTerm{
+								{
+									LabelSelector: &metav1.LabelSelector{
+										MatchExpressions: []metav1.LabelSelectorRequirement{
+											{
+												Key:      "foo",
+												Operator: metav1.LabelSelectorOpExists,
+											},
+										},
+									},
+									TopologyKey: "region",
+								},
+								{
+									LabelSelector: &metav1.LabelSelector{
+										MatchExpressions: []metav1.LabelSelectorRequirement{
+											{
+												Key:      "bar",
+												Operator: metav1.LabelSelectorOpExists,
+											},
+										},
+									},
+									TopologyKey: "zone",
+								},
+							},
+						},
+					},
+				},
+			},
+			pods: []*v1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "pod1", Labels: map[string]string{"foo": ""}},
+					Spec: v1.PodSpec{
+						NodeName: "nodeA",
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "pod2", Labels: map[string]string{"bar": ""}},
+					Spec: v1.PodSpec{
+						NodeName: "nodeB",
+					},
+				},
+			},
+			nodes: []v1.Node{
+				{ObjectMeta: metav1.ObjectMeta{Name: "nodeA", Labels: map[string]string{"region": "r1", "zone": "z1", "hostname": "nodeA"}}},
+				{ObjectMeta: metav1.ObjectMeta{Name: "nodeB", Labels: map[string]string{"region": "r1", "zone": "z2", "hostname": "nodeB"}}},
+			},
+			wantStatuses: []*framework.Status{unschedulableAndUnresolvable, unschedulableAndUnresolvable},
+			name:         "Test incoming pod's affinity: firstly check if all affinityTerms match, and then check if all topologyKeys match, and the match logic should be satified on the same pod",
+		},
+	}
+
+	for indexTest, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			nodeListInfo := predicates.FakeNodeListInfo(test.nodes)
+			nodeInfoMap := make(map[string]*schedulernodeinfo.NodeInfo)
+			for i, node := range test.nodes {
+				var podsOnNode []*v1.Pod
+				for _, pod := range test.pods {
+					if pod.Spec.NodeName == node.Name {
+						podsOnNode = append(podsOnNode, pod)
+					}
+				}
+				nodeInfo := schedulernodeinfo.NewNodeInfo(podsOnNode...)
+				nodeInfo.SetNode(&test.nodes[i])
+				nodeInfoMap[node.Name] = nodeInfo
+			}
```

Comment:
you can use the following function instead of this block:

```nodeInfoMap = schedulernodeinfo.CreateNodeNameToInfoMap(test.pods, test.nodes)```

Reply:
done